semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md

```diff
@@ -0,0 +1,202 @@
+Since Elasticsearch stores documents with a separate key/id and value, the mapper will serialize all properties except for the key to a JSON object
+and use that as the value.
+
+Usage of the `JsonPropertyNameAttribute` is supported if a different storage name to the
+data model property name is required. It is also possible to use a custom `JsonSerializerOptions` instance with a customized property naming policy. To enable this, 
+a custom source serializer must be configured.
+
+```csharp
+using Elastic.SemanticKernel.Connectors.Elasticsearch;
+using Elastic.Clients.Elasticsearch;
+using Elastic.Clients.Elasticsearch.Serialization;
+using Elastic.Transport;
+
+var nodePool = new SingleNodePool(new Uri("http://localhost:9200"));
+var settings = new ElasticsearchClientSettings(
+    nodePool,
+    sourceSerializer: (defaultSerializer, settings) =>
+        new DefaultSourceSerializer(settings, options => 
+            options.PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseUpper));
+var client = new ElasticsearchClient(settings);
+
+var collection = new ElasticsearchVectorStoreRecordCollection<Hotel>(
+    client,
+    "skhotelsjson");
+```
+
+As an alternative, the `DefaultFieldNameInferrer` lambda function can be configured to achieve the same result or to even further customize property naming based on dynamic conditions.
+
+```csharp
+using Elastic.SemanticKernel.Connectors.Elasticsearch;
+using Elastic.Clients.Elasticsearch;
+
+var settings = new ElasticsearchClientSettings(new Uri("http://localhost:9200"));
+settings.DefaultFieldNameInferrer(name => JsonNamingPolicy.SnakeCaseUpper.ConvertName(name));
+var client = new ElasticsearchClient(settings);
+
+var collection = new ElasticsearchVectorStoreRecordCollection<Hotel>(
+    client,
+    "skhotelsjson");
+```
+
+Since a naming policy of snake case upper was chosen, here is an example of how this data type will be set in Elasticsearch.
+Also note the use of `JsonPropertyNameAttribute` on the `Description` property to further customize the storage naming.
+
+```csharp
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.VectorData;
+
+public class Hotel
+{
+    [VectorStoreRecordKey]
+    public string HotelId { get; set; }
+
+    [VectorStoreRecordData(IsFilterable = true)]
+    public string HotelName { get; set; }
+
+    [JsonPropertyName("HOTEL_DESCRIPTION")]
+    [VectorStoreRecordData(IsFullTextSearchable = true)]
+    public string Description { get; set; }
+
+    [VectorStoreRecordVector(Dimensions: 4, DistanceFunction.CosineSimilarity, IndexKind.Hnsw)]
+    public ReadOnlyMemory<float>? DescriptionEmbedding { get; set; }
+}
+```
+
+```json
+{
+  "_index" : "skhotelsjson",
+  "_id" : "h1",
+  "_source" : {
+    "HOTEL_NAME" : "Hotel Happy",
+    "HOTEL_DESCRIPTION" : "A place where everyone can be happy.",
+    "DESCRIPTION_EMBEDDING" : [
+      0.9,
+      0.1,
+      0.1,
+      0.1
+    ]
+  }
+}
+```
+
+::: zone-end
```

Comment:
To fix the warnings, you'll have to add the python and java sections as well.  What we are typically doing is adding coming soon for those or whatever the appropriate message may be, e.g.

```
::: zone pivot="programming-language-python"

## Coming soon

More info coming soon.

::: zone-end
::: zone pivot="programming-language-java"

## Coming soon

More info coming soon.

::: zone-end
```

Reply:
The invalid author warnings are probably not as easy to fix. Should I just use your name or how do we handle these?

Reply:
Sure, you can use the following, thanks!
```
ms.author: westey
```